mlflow/tracking/client.py

```diff
@@ -1527,6 +1527,103 @@ def log_table(
+            :caption: Example with passing run_ids
+
+            import mlflow
+            import pandas as pd
+
+            client = MlflowClient()
+            loaded_table = client.load_table(
+                experiment_id="uuid-0",
+                artifact_file="qabot_eval_results.csv",
+                run_ids=[
+                    "68f0c4aed0b74fc7bff23d6c3be74fed",
+                    "da1f56b292254a81858410606618d61c",
+                ],
+                # Append a column containing the associated run ID for each row
+                extra_columns=["run_id"],
+            )
+
+        .. test-code-block:: python
+            :caption: Example with passing no run_ids
+
+            # Loads the table with the specified name for all runs in the given
+            # experiment and joins them together
+            loaded_table = mlflow.load_table(
+                experiment_id="uuid-0",
+                artifact_file="qabot_eval_results.csv",
+                # Append the run ID and the parent run ID to the table
+                extra_columns=["run_id", "parent_run_id"],
+            )
+        """
+        import pandas as pd
+
+        subset_tag_value = json.dumps({"path": artifact_file, "type": "table"})
+
+        # Build the filter string
+        tag_query = f"tags.{MLFLOW_LOGGED_ARTIFACTS} LIKE '%{subset_tag_value}%'"
+        run_query = ""
+        if run_ids is not None:
+            list_run_ids = ",".join(run_ids)
+            run_query = f" and attributes.run_id IN ('{list_run_ids}')"
+
+        filter_string = tag_query + run_query
+        runs = mlflow.search_runs(experiment_ids=[experiment_id], filter_string=filter_string)
+        data = pd.DataFrame()
+
+        # TODO: Add parallelism support here
+        def get_artifact_data(run):
+            run_id = run.run_id
+            norm_path = posixpath.normpath(artifact_file)
+            artifact_dir = posixpath.dirname(norm_path)
+            artifact_dir = None if artifact_dir == "" else artifact_dir
+            existing_predictions = pd.DataFrame()
+
+            artifacts = [f.path for f in self.list_artifacts(run_id, path=artifact_dir)]
+            if artifact_file in artifacts:
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    downloaded_artifact_path = mlflow.artifacts.download_artifacts(
+                        run_id=run_id, artifact_path=artifact_file, dst_path=tmpdir
+                    )
+                    existing_predictions = pd.read_json(downloaded_artifact_path, orient="split")
+                    if extra_columns is not None:
+                        for column in extra_columns:
+                            existing_predictions[column] = existing_predictions.apply(
+                                lambda x, column=column: run[column], axis=1
+                            )
+            else:
+                raise MlflowException(f"Artifact {artifact_file} not found for run {run_id}.")
```

Comment:
Can we validate all the tables are downloadable beforehand to fail fast and avoid useless downloads?

Reply:
How do we verify that if the a particular artifact is downloadable? I thought, checking if the path exists should help with making sure that the file is downloadable.

Also this else should never ever happen if the search query is accurate. I have just added it for safety sake.

Reply:
> Also this else should never ever happen if the search query is accurate. I have just added it for safety sake.

I meant something like this:

```python
for run_id in run_ids:
    artifacts = [
        f.path for f in self.list_artifacts(run_id, path=artifact_dir) if not f.is_dir
    ]
    raise MlflowException(f"Artifact {artifact_file} not found for run {run_id}.")


# Then, download them
```

> Also this else should never ever happen if the search query is accurate. I have just added it for safety sake.


I missed this.